course/views.py

```diff
@@ -1,26 +1,33 @@
-from django.shortcuts import get_list_or_404
-from django.http import JsonResponse
+from course.models import Course, CourseCategory
+from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework.decorators import detail_route
 
-from course.models import Course
-from course.models import CourseCategory
+from course.serializers import CourseSerializer, CourseCategorySerializer
 
 
-def get_catalog(request):
-    courses = list(Course.objects.values())
-    return JsonResponse({
-        "data": courses
-    })
+class CourseViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows courses to be viewed or edited
+    """
+    queryset = Course.objects.all()
+    serializer_class = CourseSerializer
 
 
-def get_course_categories(request):
-    course_categories = list(CourseCategory.objects.values())
-    return JsonResponse({
-        "data": course_categories
-    })
+class CourseCategoryViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows course categories to be viewed or edited
+    """
+    queryset = CourseCategory.objects.all()
+    serializer_class = CourseCategorySerializer
 
-
-def get_courses_for_category(request, category_id):
-    course_list = get_list_or_404(Course, course_category=category_id)
-    return JsonResponse({
-        "data": course_list
-    })
+    """
+    Extra route that gets courses by category id
```

Comment:
This method of adding a custom route may not be officially supported anymore - saw it in a documentation site for "Django Rest Framework 2", will look into other ways to achieve the same thing based on regular site but seems to work fine for now

Reply:
alternative seems to be using "action" decorator, will test out on my local repo and see how it works